x-pack/plugins/data_enhanced/server/search/es_search_strategy.ts

```diff
@@ -28,15 +28,17 @@ export interface AsyncSearchResponse<T> {
   response: SearchResponse<T>;
 }
 
-export const enhancedEsSearchStrategyProvider: TSearchStrategyProvider<typeof ES_SEARCH_STRATEGY> = (
-  context: ISearchContext,
-  caller: APICaller
-) => {
+export const enhancedEsSearchStrategyProvider = async (
+  config$: Observable<SharedGlobalConfig>
+): Promise<ISearchStrategy<typeof ES_SEARCH_STRATEGY>> => {
+  const config = await config$.pipe(first()).toPromise();
+
   const search: ISearch<typeof ES_SEARCH_STRATEGY> = async (
+    context,
     request: IEnhancedEsSearchRequest,
     options
   ) => {
-    const config = await context.config$.pipe(first()).toPromise();
+    const caller = context.core.elasticsearch.legacy.client.callAsCurrentUser;
```

Comment:
//nit 
Should we maybe pass atleast the elasticsearch sub service into the context, instead of passing entire core?